### ISSUE TITLE
[🐞] `useTask$` doesn't behave the same with Link navigation

### ISSUE THREAD

**Issue by shairez** [BuilderIO/qwik#4089]
### Which component is affected?

Qwik Runtime

### Describe the bug

AFAIK, `useTask$` should run in the same deterministic way whether it's server rendered or client rendered.

But it doesn't run in the same order when it runs during client side rendering (using `Link` or "show/hide" logic)

### Reproduction

https://stackblitz.com/edit/qwik-starter-hjsrfx?file=src/routes/page1/tabs.tsx

### Steps to reproduce

1. Click on `Page 1 - Go here first`
![image](https://user-images.githubusercontent.com/1430726/236648932-6bcf7fe7-1e05-479c-b3b2-a63938e90bfa.png)

2. Look at the right order of tabs index numbers -
![image](https://user-images.githubusercontent.com/1430726/236649063-eaef4a18-6042-4cf4-b799-49d426883a89.png)

3. Click on the link `Page 2 - Click Here`

![image](https://user-images.githubusercontent.com/1430726/236649076-7d80f886-71b9-4a27-ac67-ce4421183fbc.png)

4. Click on the only link on the page 
 
![image](https://user-images.githubusercontent.com/1430726/236649082-0db7b3f2-6d34-4da5-a7a6-cdc369425302.png)

5. Look at the wrong order of index numbers
![image](https://user-images.githubusercontent.com/1430726/236648972-57e54867-9335-4493-b2c7-139c5f189418.png)

6. The "problematic code" is located the `page1/tabs.tsx` file, just search for `useTask$`
(If I change it to `useVisibleTask$`, it works, but with a flicker, obviously)

### System Info

```shell
(From Stackblitz)

System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @builder.io/qwik: 1.0.0 => 1.0.0 
    @builder.io/qwik-city: ^1.0.0 => 1.0.0 
    undici: 5.22.0 => 5.22.0 
    vite: 4.3.3 => 4.3.3
```

### Additional Information

The "problematic code" is located the `page1/tabs.tsx` file, just search for `useTask$`
(If I change it to `useVisibleTask$`, it works, but with a flicker, obviously)

**Comment by manucorporat** [comment]
sibling run in parallel, in SSR is deterministic because there is not lazy loading

**Comment by shairez** [comment]
Thanks
This problem doesn't occur when using `useVisibleTask$` by the way, but that comes with an annoying "flicker"

Do you have any suggestion of how we could solve the problem of syncing up sibling child components? (like in this case, tabs + their content)

**Few options we thought of -** 
1. Having it not behave in parallel on the browser (make it the same as in SSR)

2. Adding a `value` prop (only in our library) where both components (tab + content) that point to the same unique value (which will work but wil make the API more verbose)

3. Having some kind of an API to query child components (like we discussed in the past, a hook that gives you like an observer of project child components), that way we'd know their actual index on the DOM.

4. using querySelector (but that will require `useVisibleTask$` again)

**Comment by manucorporat** [comment]
it also executes in parallel in SSR, but there are no promises, no lazy loading. Looking into some code to make this more determinisitc in cases like this

**Comment by shairez** [comment]
OK thanks @manucorporat ! 🙏

**Comment by wmertens** [comment]
Not sure this should be open - isn't this by design? You can't count on the rendering order, you should use the signal dependency graph instead.

**Comment by Dindaleon** [comment]
I am having similar issues. Everything works fine whenever I load the page; however, when I go to the page through a Link, the loading order is all messed up and the page won't load correctly.

**Comment by wmertens** [comment]
@Dindaleon depends, can you share some code

**Comment by wmertens** [comment]
@Dindaleon That's not possible if a component stays mounted. The order of task execution is only in code-order the first render, after that it depends on tracked signals.

So unless you're seeing incorrect ordering in newly mounted components, it's expected behavior.

I'm closing this issue as it's behaving as expected, but if you do find a bug please open a new issue with a minimal reproduction.

**Comment by wmertens** [comment]
BTW - the solution for the ordering of tabs in Qwik UI is using an Inline Component, which has access to the children, and providing them with index props before passing them to the Qwik Component.
